tests/test_hierarchy.cc

```diff
@@ -18,6 +18,8 @@
 #include <mfmg/dealii_adapters.hpp>
 #include <mfmg/hierarchy.hpp>
 
+#include <test_hierarchy_helpers.hpp>
+
```

Comment:
Is it the reason why you want to include the `tests` directory in CMake? You can just use `#include "test_hierarchy_helpers.hpp"` instead. That's what we do with `laplace.hpp`